src/generate/generic.rs

```diff
@@ -47,6 +50,12 @@ pub trait RegisterSpec {
     type Ux: RawReg;
 }
 
+/// Raw field type
+pub trait FieldSpec: Sized {
+    /// Raw field type (`u8`, `u16`, `u32`, ...).
+    type Ux: Copy + PartialEq + From<Self>;
+}
+
```

Comment:
why not make this a supertrait?

```suggestion
/// Raw field type
pub trait FieldSpec: Copy + PartialEq + From<Self> + Sized {}

```

Reply:
`Self` and `Ux` are different types if field has enumeratedValues.